From ff3eb7b97b0c6c8fe8d775be73832720e2006dc6 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Thu, 10 Sep 2020 17:34:52 +0200
Subject: [PATCH 103/134] adaptive: Move merging/dumping out of gddv parsing

GDDV parsing needs to be recurive, so move out this common code to
simplify calling it recursively.

(cherry picked from commit 7aba2da2b5cf5ed7ae2cd1fd3d2f87d716d41c84)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
(cherry picked from commit 74aa62091fd3bfb47062b06bfa69a907111beec3)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 src/thd_engine_adaptive.cpp | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/thd_engine_adaptive.cpp b/src/thd_engine_adaptive.cpp
index 7d79de8..9f791ff 100644
--- a/src/thd_engine_adaptive.cpp
+++ b/src/thd_engine_adaptive.cpp
@@ -698,13 +698,6 @@ int cthd_engine_adaptive::parse_gddv(char *buf, int size) {
 		delete[] (val);
 	}
 
-	merge_appc();
-
-	dump_ppcc();
-	dump_psvt();
-	dump_apat();
-	dump_apct();
-
 	return 0;
 }
 
@@ -1398,6 +1391,13 @@ int cthd_engine_adaptive::thd_engine_start(bool ignore_cpuid_check) {
 			delete[] buf;
 			return THD_ERROR;
 		}
+
+		merge_appc();
+
+		dump_ppcc();
+		dump_psvt();
+		dump_apat();
+		dump_apct();
 	} catch (std::exception &e) {
 		thd_log_warn("%s\n", e.what());
 		delete [] buf;
-- 
2.27.0

